Additional THM sqa items

GiudGiud opened the pull request:
I had some free time in the plane.
This is only something like 10% of the tests (about 280 of them concerned) that need it but same as last time, we'll see the end of it

EDIT: ended up getting all the others in a different PR
That other PR will have to deal with a larger issue (non-matching commits & issues) so I would rather this one be processed independently

GiudGiud committed:
Add SQA testing documentation to THM peacock integration
Add supporting documentation

GiudGiud committed:
Add SQA testing data for thm auxkernels

GiudGiud committed:
Add SQA testing documentation to closures and simulation classes

GiudGiud committed:
Add some components testing metadata for sqa, refs #26336

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1946275/) on 481ac7f wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/26338/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1946234/) on 481ac7f wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26338/site)

This comment will be updated on new commits.

joshuahansel commented on modules/thermal_hydraulics/test/tests/auxkernels/sound_speed/tests:
```diff
@@ -1,7 +1,12 @@
 [Tests]
+  # see relap-7#636
+  # r26174
```
typo?

GiudGiud replied:
no this is copied from the commit.

joshuahansel replied:
I just don't know what "r26174" is? Is the "r" supposed to be a "#"?

GiudGiud replied:
Well me neither. And I doubt at the time our issues were is the 26000s. I ll investigate maybe they also mistyped the number

GiudGiud replied:
came from this
```
    Adding pressure based stabilization for single phase
    
    Author: David Andrs <david.andrs@inl.gov>
    Signed-off-by: Richard Martineau <richard.martineau@inl.gov>
    
    r26174
```

joshuahansel replied:
Since its meaning has been lost

GiudGiud replied:
works for me, it s consistent with the approach for the others

joshuahansel commented on modules/thermal_hydraulics/test/tests/base/simulation/tests:
```diff
@@ -17,12 +20,14 @@
     input = 'err.no_smp.i'
     expect_err = "Single matrix preconditioning with full coupling is required to run\. Please, check that your input file has the following preconditioning block\:"
     deleted = "#190"
+    requirement = 'The system shall error if single matrix preconditioning is not in use in a thermal hydraulics simulation.'
```
I think we should do "report an error" instead of "error", since "error" sounds like something you don't want your system to do :) Same comment for your other requirements

GiudGiud replied:
interesting both are used approximately in the same quantity
```
(moose) ~/projects/moose_v1/modules$ ggrep -r "The system shall report an error" | wc -l
      39
(moose)~/projects/moose_v1/modules$ ggrep -r "The system shall error" | wc -l
      34
(moose) ~/projects/moose_v1/modules$ cd ../test/
(moose) ~/projects/moose_v1/test$ ggrep -r "The system shall error" | wc -l
^[[A^[[A^[[A     142
(moose) ~/projects/moose_v1/test$ ggrep -r "The system shall report an error" | wc -l
     140
(moose) ~/projects/moose_v1/test$ 
```

GiudGiud replied:
I can change though

joshuahansel replied:
Yeah "report an error" is preferable going forward

GiudGiud replied:
alright well I changed the repo so we do not go backward

GiudGiud committed:
Address Josh's review
- use the system shall report an error instead of shall error
Apply everywhere so we use the correct one moving forward

GiudGiud commented:
Thanks for the review! now for the big one haha. (no need to look now it s not ready)
